guvnor-m2repo-editor/guvnor-m2repo-editor-client/src/main/java/org/guvnor/m2repo/client/upload/UploadFormViewImpl.java

```diff
@@ -195,8 +146,80 @@ private String getWebContext() {
         return context;
     }
 
-    protected void showUploadingBusy() {
-        // LoadingPopup.showMessage( "Uploading...");
+    public void showUploadingBusy() {
+        // TODO: "Uploading..." needs localization
+        BusyPopup.showMessage("Uploading...");
+    }
+
+    public void hideUploadingBusy() {
+        BusyPopup.close();
+    }
+
+    @Override
+    public void showWarning(String message) {
+        Window.alert(message);
+    }
+
+    @Override
+    public void showSelectFileUploadWarning() {
+        showWarning(M2RepoEditorConstants.INSTANCE.SelectFileUpload());
+    }
+
+    @Override
+    public void showUnsupportedFileTypeWarning() {
+        showWarning(M2RepoEditorConstants.INSTANCE.UnsupportedFileType());
     }
 
+    @Override
+    public void showUploadedSuccessfullyWarning() {
+        showWarning(M2RepoEditorConstants.INSTANCE.UploadedSuccessfully());
+    }
+
+    @Override
+    public void showInvalidJarNotPomWarning() {
+        showWarning(M2RepoEditorConstants.INSTANCE.InvalidJarNotPom());
+    }
+
+    @Override
+    public void showInvalidPomWarning() {
+        showWarning(M2RepoEditorConstants.INSTANCE.InvalidPom());
+    }
+
+    @Override
+    public void showUploadFailedError(String message) {
+        showErrorMessage(M2RepoEditorConstants.INSTANCE.UploadFailed() + message);
+    }
+
+    @Override
+    public void showGAVInputs() {
+        hiddenFieldsPanel.setVisible(true);
+    }
+
+    @Override
+    public void hideGAVInputs() {
+        hiddenFieldsPanel.setVisible( false );
+        hiddenArtifactIdField.setText( null );
+        hiddenGroupIdField.setText( null );
+        hiddenVersionIdField.setText( null );
+    }
+
+    @Override
+    public String getFileName() {
+        return uploader.getFilename();
+    }
+
+    @Override
+    public void fireSearchEvent() {
+        searchEvent.fire(new M2RepoSearchEvent());
```

Comment:
@manstis +1. For that the presenter needs to be turned into a bean (which it currently is not) and then, in M2 perspective, obtained via iocManager so that injection is performed. Do I understand that correctly?